src/Command/AllCommand.php

```diff
@@ -98,7 +101,13 @@ public function execute(Arguments $args, ConsoleIo $io): ?int
             /** @var \Cake\Command\Command $command */
             $command = new $commandName();
             foreach ($tables as $table) {
-                $subArgs = new Arguments([$table], $args->getOptions(), ['name']);
+                $options = $args->getOptions();
+                if ($command instanceof ModelCommand) {
+                    if ($args->hasOption('prefix')) {
+                        unset($options['prefix']);
+                    }
+                }
```

Comment:
```suggestion
                if (
                    $args->hasOption('prefix') &&
                    !($command instanceof ControllerCommand) &&
                    !($command instanceof TemplateCommand)
                ) {
                    unset($options['prefix']);
                }
```

It might be better to apply the option only to the commands that are supported, instead of not applying to those that aren't supported, this means less possible failures further down the road (for example in case additional commands are being added).

You can also put it outside the `foreach` loop, as it only needs to be checked once per command.